### PR TITLE
Create a responsive view state to ease implementation on flutter web

### PR DESCRIPTION
**Pull request by rafaelcmm**
## Description

*This PR creates a ResponsiveViewState, to abstract responsiveness of flutter_clean_architecture view*

## What was done
- Created `ResponsiveViewState` abstract class to ease web development with flutter and improve its support.
- Removed circleci steps to use Github Actions
- Improved lint and formatter

## Developer checklist

- [x] All github actions are passing.

## For reviewers

All PR's needs at least one reviewer to be merged.

Before merge this PR confirm that it meets all requirements listed below:

- The PR has good quality code and has tests (if is the case)
- This PR is not checked with WIP on title

### CODE REVIEW

File: example/lib/src/data/repositories/data_users_repository.dart [hunk 1]
```diff
@@ -4,9 +4,9 @@ import '../../domain/repositories/users_repository.dart';
 class DataUsersRepository extends UsersRepository {
   List<User> users;
   // sigleton
-  static DataUsersRepository _instance = DataUsersRepository._internal();
+  static final DataUsersRepository _instance = DataUsersRepository._internal();
   DataUsersRepository._internal() {
-    users = List<User>();
+    users = <User>[];
```
Comment (ShadyBoukhary) [inline example/lib/src/data/repositories/data_users_repository.dart:9]: Is there a benefit to this change?

File: lib/src/view.dart [hunk 2]
```diff
@@ -4,6 +4,130 @@ import 'package:logging/logging.dart';
...
+      return ScreenSizeType.TABLET;
+    }
+
+    return ScreenSizeType.DESKTOP;
+  }
+
+  /// This turns buildPage into an implicit method that build according to the given builds methods: [MOBILE], [TABLET] and [DESKTOP].
+  /// The Default Viewport is [MOBILE]. When [TABLET] or [DESKTOP] builds are null, [MOBILE] viewport will be called. If all the build are null,
+  /// it will throw an [UnimplentedError].
+  @override
```
Comment (ShadyBoukhary) [inline lib/src/view.dart:102]:
```suggestion
  @override
  @nonvirtual
```

File: lib/src/view.dart [hunk 3]
```diff
@@ -4,6 +4,130 @@ import 'package:logging/logging.dart';
...
+            case ScreenSizeType.DESKTOP:
+              return buildDesktopView() ??
+                  buildTabletView() ??
+                  buildMobileView();
+            default:
+          }
+        } catch (e) {
+          print(e);
+        }
+        throw UnimplementedError('Implement at least one build method');
```
[the remark this answers is not in the document]
Comment (rafaelcmm) [reply lib/src/view.dart:125]: Sure, no problem

File: example/lib/src/data/repositories/data_users_repository.dart (hunk 1 above)
Comment (rafaelcmm) [reply example/lib/src/data/repositories/data_users_repository.dart:9]: As defined by new lint rules in `pedantic 1.9.0`, we can omit types in this lists to improve legibility

**Pull request merged**